verification/curator-service/ui/src/components/AutomatedSourceForm.tsx

```diff
@@ -0,0 +1,192 @@
+import * as Yup from 'yup';
+
+import { Button, Typography, makeStyles } from '@material-ui/core';
+import { FastField, Form, Formik } from 'formik';
+
+import AppModal from './AppModal';
+import FieldTitle from './common-form-fields/FieldTitle';
+import MuiAlert from '@material-ui/lab/Alert';
+import React from 'react';
+import { RequiredHelperText } from './common-form-fields/FormikFields';
+import { TextField } from 'formik-material-ui';
+import User from './User';
+import axios from 'axios';
+import { useHistory } from 'react-router-dom';
+
+const useStyles = makeStyles(() => ({
+    headerBlurb: {
+        maxWidth: '70%',
+        paddingBottom: '3em',
+        paddingTop: '1em',
+    },
+    headerText: {
+        marginTop: '2em',
+    },
+    fieldTitle: {
+        paddingBottom: '1em',
+    },
+    form: {
+        display: 'flex',
+        flexDirection: 'column',
+        height: '100%',
+        paddingLeft: '3em',
+        paddingRight: '4em',
+    },
+    formSection: {
+        paddingBottom: '2em',
+    },
+    allFormSections: {
+        marginBottom: '2em',
+        maxWidth: '60%',
+        paddingLeft: '2em',
+        paddingRight: '1em',
+        paddingTop: '0.5em',
+    },
+    statusMessage: {
+        marginTop: '2em',
+        maxWidth: '80%',
+    },
+    uploadBar: {
+        alignItems: 'center',
+        display: 'flex',
+        height: '4em',
+        marginTop: 'auto',
+    },
+    cancelButton: {
+        marginLeft: '1em',
+    },
+}));
+
+interface Props {
+    user: User;
+    onModalClose: () => void;
+}
+
+export interface AutomatedSourceFormValues {
+    url: string | null;
+    name: string | null;
+}
+
+const AutomatedSourceFormSchema = Yup.object().shape({
+    url: Yup.string().nullable().required('Required'),
+    name: Yup.string().nullable().required('Required'),
+});
+
+export default function AutomatedSourceForm(props: Props): JSX.Element {
+    const classes = useStyles();
+    const history = useHistory();
+    const [errorMessage, setErrorMessage] = React.useState('');
+
+    const createSource = async (
+        values: AutomatedSourceFormValues,
+    ): Promise<void> => {
+        const newSource = { name: values.name, origin: { url: values.url } };
+        try {
+            await axios.post('/api/sources', newSource);
+            setErrorMessage('');
+        } catch (e) {
+            setErrorMessage(JSON.stringify(e));
+            return;
+        }
+        // Navigate to sources after successful submit
+        history.push({
+            pathname: '/sources',
+        });
+    };
+
+    return (
+        <AppModal
+            title="New automated data source"
+            onModalClose={props.onModalClose}
+        >
+            <Formik
+                validationSchema={AutomatedSourceFormSchema}
+                validateOnChange={false}
+                initialValues={{ url: null, name: null }}
```

Comment:
When you start typing in the inputs are there console errors about changing from an uncontrolled to controlled component? That happens when the initial values are undefined, but I can't remember if it happens when the initial values are null. If the errors do appear, you can change these to `''` rather than `null`.

Reply:
It does, thanks for the catch. This doesn't quite interoperate with `RequiredHelperText`, so I'll take that out (this means we won't have `Required` visible if there isn't an error, but given the size of this form, that's probably fine).